### ISSUE TITLE
[ENH]: Multi Output Regression and Classification

### ISSUE THREAD

**Issue by moezali1** [pycaret/pycaret#2512]
### Describe the feature you want to add to this project

Is it possible to make PyCaret 3.0 compatible for Multi-Output Regressor and Classifier (>2 targets)?

![image](https://user-images.githubusercontent.com/54699234/167178283-321b0a15-f95b-4506-8901-ef22d5e1bb33.png)

We can do so by accepting the list of strings as the `target` parameter in the setup. I am not sure how we will handle metric evaluation. Might need to explore further.

### Describe your proposed solution

See above

**Comment by Yard1** [comment]
Definetely not for RC1

**Comment by ngupta23** [comment]
Yes, it is not such a trivial change and definitely not in 3.0.0. when we add this, we should simultaneously think about how to add multivariate time series support which supports multiple targets.

**Comment by ngupta23** [comment]
This will have an impact on plotting as well. How would this be handled?

**Comment by tvdboom** [comment]
I agree with @ngupta23 and @Yard1. This is quite a bigger change than it may seem. I think it would be a good enhancement to the package but let's focus on it after 3.0 is out and bug free.

**Comment by moezali1** [comment]
Ok I agree with all of you. I will put block on this.

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
